packages/matter.js/src/device/PairedNode.ts

```diff
@@ -229,29 +307,56 @@ export class PairedNode {
     }
 
     get isConnected() {
-        return this.connectionState === NodeStateInformation.Connected;
+        return this.#connectionState === NodeStateInformation.Connected;
     }
 
     get nodeState() {
-        return this.connectionState;
+        return this.#connectionState;
     }
 
     get basicInformation() {
         return this.#nodeDetails.basicInformation;
     }
 
+    get localInitializationDone() {
+        return this.#localInitializationDone;
+    }
+
+    get remoteInitializationDone() {
+        return this.#remoteInitializationDone;
+    }
+
+    get initialized() {
+        return this.#remoteInitializationDone || this.#localInitializationDone;
+    }
+
     private setConnectionState(state: NodeStateInformation) {
         if (
-            this.connectionState === state ||
-            (this.connectionState === NodeStateInformation.WaitingForDeviceDiscovery &&
+            this.#connectionState === state ||
+            (this.#connectionState === NodeStateInformation.WaitingForDeviceDiscovery &&
                 state === NodeStateInformation.Reconnecting)
         )
             return;
-        this.connectionState = state;
+        this.#connectionState = state;
         this.options.stateInformationCallback?.(this.nodeId, state);
         this.events.nodeStateChanged.emit(state);
         if (state === NodeStateInformation.Disconnected) {
-            this.reconnectDelayTimer.stop();
+            this.#reconnectDelayTimer?.stop();
+            this.#reconnectDelayTimer = undefined;
+        }
+    }
+
+    /** Make sure to not request a new Interaction client multiple times in parallel. */
+    async handleReconnect(discoveryType?: NodeDiscoveryType): Promise<void> {
+        if (this.#clientReconnectInProgress) {
+            throw new NodeNotConnectedError("Reconnection already in progress. Node not reachable currently.");
```

Comment:
Should we return the original reconnect promise?  Assuming multiple processes are waiting for reconnect...  Then they'd both wake up when the connection is ready

If there should only ever be one interested party then maybe this should be an internal error

Reply:
This method is called by the "timer based reconnection process" internally in most cases, so here it is already made sure that all is good.

But the other way to trigger that would be when an interaction Client is requested by the user to e.g. send commands directly or such ... in this case I decided it is better to directly return an error, else the call could wait "indefinitively" (because device is being discovered). In fact these guards are there to make sure that in worst case errors are indeed returned to the dev - so better more then waiting unlimited amount of time and blocking.
Formally currently the PairedNode do not know if we try to reconnect or if we wait for discovery already.